src/webgl/material.js

```diff
@@ -513,6 +513,112 @@ function material(p5, fn){
     p5._validateParameters('createShader', arguments);
     return new Shader(this._renderer, vertSrc, fragSrc, options);
   };
+  /**
+   * Creates and loads a filter shader from an external file.
+   *
+   * @method loadFilterShader
+   * @param {String} fragFilename path to the fragment shader file
+   * @param {Function} [successCallback] callback to be called once the shader is
+   *                                     loaded. Will be passed the
+   *                                     <a href="#/p5.Shader">p5.Shader</a> object.
+   * @param {Function} [failureCallback] callback to be called if there is an error
+   *                                     loading the shader. Will be passed the
+   *                                     error event.
+   * @return {Promise<p5.Shader>} a promise that resolves with a shader object
+   *
+   * @example
+   * <div modernizr='webgl'>
+   * <code>
+   * let myShader;
+   *
+   * async function setup() {
+   *    myShader = await loadFilterShader('assets/shader.frag');
+   *   createCanvas(100, 100, WEBGL);
+   *   noStroke();
+   * }
+   *
+   * function draw() {
+   *   // shader() sets the active shader with our shader
+   *   shader(myShader);
+   *
+   *   // rect gives us some geometry on the screen
+   *   rect(0, 0, width, height);
+   * }
+   * </code>
+   * </div>
+   * @alt
+   * A rectangle with a shader applied to it.
+   */
+  p5.prototype.loadFilterShader = async function (fragFilename, successCallback, failureCallback) {
+    p5._validateParameters('loadFilterShader', arguments);
+
+    const loadedShader = new p5.Shader();
+
+    try {
+      // Define the default vertex shaders for different WebGL versions
+      const defaultVertV1 = `
```

Comment:
Thanks, this is looking good!

Is this bit of code repeated from `createFilterShader`? I wonder if there's a way where we can reuse the same code rather than duplicating it. Maybe you could load the file first, and then pass the code directly into `createFilterShader` and return the results?

Reply:
Sorry I overlooked yesterday , today I noticed when I was testing their was a .copyToContext error when reusing createFilterShader ,I've fixed that too now